test/lib/ufe/testParentCmd.py

```diff
@@ -0,0 +1,222 @@
+        cylHier = ufe.Hierarchy.hierarchy(cylinderItem)
+        cylChildren = cylHier.children()
+        self.assertEqual(len(cylChildren), 0)
+
+        # Get the components of the cube's local transform.
+        cubeT3d = ufe.Transform3d.transform3d(cubeItem)
+        cubeT = cubeT3d.translation()
+        cubeR = cubeT3d.rotation()
+        cubeS = cubeT3d.scale()
+
+        # Parent cube to cylinder in relative mode, using UFE path strings.
+        cmds.parent("|mayaUsdProxy1|mayaUsdProxyShape1,/pCube1",
+                    "|mayaUsdProxy1|mayaUsdProxyShape1,/pCylinder1",
+                    relative=True)
+
+        # Confirm that the cube is now a child of the cylinder.
+        cylChildren = cylHier.children()
+        self.assertEqual(len(cylChildren), 1)
+        self.assertIn("pCube1", childrenNames(cylChildren))
+
+        # Undo: the cube is no longer a child of the cylinder.
+        cmds.undo()
+
+        cylChildren = cylHier.children()
+        self.assertEqual(len(cylChildren), 0)
+
+        # Redo: confirm that the cube is again a child of the cylinder.
+        cmds.redo()
+
+        cylChildren = cylHier.children()
+        self.assertEqual(len(cylChildren), 1)
+        self.assertIn("pCube1", childrenNames(cylChildren))
+
+        # Confirm that the cube's local transform has not changed.  Must
+        # re-create the item, as its path has changed.
+        cubeChildPath = ufe.Path(
+            [shapeSegment, usdUtils.createUfePathSegment("/pCylinder1/pCube1")])
+        cubeChildItem = ufe.Hierarchy.createItem(cubeChildPath)
+        cubeChildT3d = ufe.Transform3d.transform3d(cubeChildItem)
+
+        self.assertEqual(cubeChildT3d.translation(), cubeT)
+        self.assertEqual(cubeChildT3d.rotation(),    cubeR)
+        self.assertEqual(cubeChildT3d.scale(),       cubeS)
+
+        # Move the parent
+        ufe.GlobalSelection.get().append(cylinderItem)
+
+        cmds.move(0, 10, 0, relative=True)
+
+        # Do the same on the equivalent Maya objects.
+        cmds.parent("pCube1", "pCylinder1", relative=True)
+        cmds.move(0, 10, 0, "pCylinder1", relative=True)
+
+        # Get its world space transform.
+        cubeWorld = cubeChildT3d.inclusiveMatrix()
+        cubeWorldList = matrixToList(cubeWorld)
+
+        # Do the same for the equivalent Maya object.
+        mayaCubeWorld = cmds.xform(
+            "|pCylinder1|pCube1", q=True, ws=True, m=True)
+
+        # Equivalent Maya and USD objects must have the same world transform. 
+        assertVectorAlmostEqual(self, cubeWorldList, mayaCubeWorld)
+
+        # Undo to bring scene to its original state.
+        for i in range(4):
+            cmds.undo()
+
+        cylChildren = cylHier.children()
+        self.assertEqual(len(cylChildren), 0)
+
+    @unittest.skipUnless(mayaUtils.previewReleaseVersion() >= 115, 'Parent functionality only available in Maya Preview Release 115 or later.') 
+    def testParentAbsolute(self):
```

Comment:
Test parent -absolute.  This changes the object's local transformation, but keeps its world transformation unchanged.  This means that undo must restore the local transformation.